### PR TITLE
[Improve][Doc] Add key features in connector documents

### LINKED ISSUE #2612: [Discuss][Doc] Unify some contents in source and sink connector documents
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The connector has some very important features, such as `batch` and `stream` and `exact-once` of the source connector. `exact-once` of sink connector. But there is no uniform fixed format description for these  in the connector-v2 document.

I suggest that we provide a fixed document template to standardize the description of these features.

Such as:
template for Source connector:

- [x] batch
- [ ] stream
- [x] exactly-once

template for Sink connector:
- [x] exactly-once

Do you have any suggestions for this?

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EricJoy2048**
fix https://github.com/apache/incubator-seatunnel/issues/2612

### CODE REVIEW

**Comment by EricJoy2048** [comment]
@hailin0  @2013650523  @TyrantLucifer  @iture123  @laglangyue  @ic4y  @legendtkl  @lhyundeadsoul  @wuchunfu  @getChan  @zhuangchong  please have a look.

**Comment by laglangyue** [comment]
Maybe we can add a little explanation or link to these words，because some people I meet don't fully understand these terms Or call it differently.
maybe it can be in Chapter Concept.

**Comment by EricJoy2048** [comment]
> Maybe we can add a little explanation or link to these words，because some people I meet don't fully understand these terms Or call it differently. maybe it can be in Chapter Concept.

Yes, I agree with you. Where shoud these explanations be put? I don't recommend that each document explain these keywords?

**Comment by TyrantLucifer** [comment]
> 

I think we can add a commom doc named `Intro to connector v2 features` to explain these concepts in the chapter as the following image shown:

image

**Comment by Hisoka-X** [comment]
> > 
> 
> I think we can add a commom doc named `Intro to connector v2 features` to explain these concepts in the chapter as the following image shown:
> 
> image

Good idea.

**Comment by EricJoy2048** [comment]
> Intro to connector v2 features

Ok.

**Comment by TyrantLucifer** [comment]
> > Intro to connector v2 features
> 
> Ok.

I see you add key-features.md in sink and source folder, I think this is also a good idea to explain the key concept.

We can add more exciting features like `support user-defined split source` or the difference between connector v1 and v2 in `Intro to connector v2 features`. What do you think about it?

**Comment by EricJoy2048** [comment]
> 

Yes, It's a good idea. I will add `support user-defined split source` to the file.

File: docs/en/concept/connector-v2-features.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+In SeaTunnel, we can save the read **Split** and its **offset**(The position of the read data in split at that time,
+such as line number, byte size, offset, etc) as **StateSnapshot** when checkpoint. If the task restarted, we will get the last **StateSnapshot**
+and then locate the **Split** and **offset** read last time and continue to send data downstream.
+
+For example `File`, `Kafka`.
+
+### schema projection
+
+If the source connector supports selective reading of certain columns or supports the data format read through `schema` params, we think it supports schema projection.
```
Comment (hailin0) [inline docs/en/concept/connector-v2-features.md]: Add `redefine columns order` description?

File: docs/en/concept/connector-v2-features.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Batch Job Mode, The data read is bounded and the job will stop when all data read complete.
+
+### stream
+
+Streaming Job Mode, The data read is unbounded and the job never stop.
+
+### parallelism
+
+Parallelism Source Connector support config `parallelism`, every parallelism will create a task to read the data.
```
Comment (hailin0) [inline docs/en/concept/connector-v2-features.md]: Add `Enumerator` and `Split` description?

**Comment by EricJoy2048** [comment]
@TyrantLucifer  @CalvinKirs  @Hisoka-X  @laglangyue  I finished it. Please have a look again.

File: docs/en/concept/connector-v2-features.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+When any piece of data flows into a distributed system, if the system processes any piece of data accurately only once in the whole processing process and the processing results are correct, it is considered that the system meets the exact once consistency.
+
+For sink connector, the sink connector supports exactly-once if any piece of data only write into target once. There are generally two ways to achieve this:
+
+* The target database supports key deduplication. For example `MySQL`, `Kudu`.
+* The target support **XA Transaction**(This transaction can be used across sessions. Even if the program that created the transaction has ended, the newly started program only needs to know the ID of the last transaction to resubmit or roll back the transaction). Then we can use **Two-phase Commit** to ensure **exactly-once**. For example `File`, `MySQL`.
+
+### schema projection
+
+If a sink connector supports the fields and their types written in the configuration, we think it supports schema projection.
```
Comment (hailin0) [inline docs/en/concept/connector-v2-features.md]: Add `redefine columns order` description?

File: docs/en/connector-v2/sink/Datahub.md [hunk 4]
```diff
@@ -6,6 +6,12 @@
 
 A sink plugin which use send message to datahub
 
+## Key features
+
+- [ ] [exactly-once](../../concept/connector-v2-features.md)
+
```
Comment (hailin0) [inline docs/en/connector-v2/sink/Datahub.md]: Some places have blank lines and others don't

File: docs/en/connector-v2/sink/Clickhouse.md [hunk 5]
```diff
@@ -4,7 +4,15 @@
 
 ## Description
 
-Used to write data to Clickhouse. Supports Batch and Streaming mode.
+Used to write data to Clickhouse.
+
+## Key features
+
```
Comment (hailin0) [inline docs/en/connector-v2/sink/Clickhouse.md:10]: Some places have blank lines and others don't

File: docs/en/concept/connector-v2-features.md (hunk 2 above)
Comment (TyrantLucifer) [reply docs/en/concept/connector-v2-features.md:42]: Agree with you.

File: docs/en/concept/connector-v2-features.md [hunk 6]
```diff
@@ -1,7 +1,18 @@
...
+# Intro To Connector V2 Features
+
+## Differences Between Connector V2 And Connector v1
+
+Connector V2 is a connector defined based on the Seatunnel Connector API interface. Unlike Connector V1, Connector V2 supports the following features.
+
+* **Multi Engine Support** SeaTunnel Connector API is an engine independent API. The connectors developed based on this API can run in multiple engines. Currently, Flink and Spark are supported, and we will support other engines in the future.
+* **Multi Engine Version Support** Decoupling the connector from the engine through the translation layer solves the problem that most connectors need to modify the code in order to support a new version of the underlying engine.
+* **Unified Batch And Stream** Connector V2 can perform batch processing or streaming processing. We do not need to develop connectors for batch and stream separately.
+* **Multiplexing JDBC/Log connection.** Connector V2 supports JDBC resource reuse and sharing database log parsing.
```
Comment (TyrantLucifer) [inline docs/en/concept/connector-v2-features.md:11]: How about add the design ideas about connector V2 api?

File: docs/en/concept/connector-v2-features.md (hunk 2 above)
Comment (EricJoy2048) [reply docs/en/concept/connector-v2-features.md]: done

File: docs/en/concept/connector-v2-features.md (hunk 3 above)
Comment (EricJoy2048) [reply docs/en/concept/connector-v2-features.md]: done

File: docs/en/connector-v2/sink/Datahub.md (hunk 4 above)
Comment (EricJoy2048) [reply docs/en/connector-v2/sink/Datahub.md]: done

File: docs/en/connector-v2/sink/Clickhouse.md (hunk 5 above)
Comment (EricJoy2048) [reply docs/en/connector-v2/sink/Clickhouse.md:10]: done

File: docs/en/concept/connector-v2-features.md (hunk 1 above)
Comment (EricJoy2048) [reply docs/en/concept/connector-v2-features.md]: done

File: docs/en/concept/connector-v2-features.md (hunk 6 above)
Comment (EricJoy2048) [reply docs/en/concept/connector-v2-features.md:11]: done

**Comment by laglangyue** [comment]
I took a cursory look，it's ok.

**Comment by hailin0** [comment]
LGTM

**Review by TyrantLucifer — APPROVED** [review]
LGTM.

**Pull request merged**